### PR TITLE
Working on the output strings

### PR DESCRIPTION
**Pull request by nsomar**
- Adding colors using swiftline
- Updating tests

Note:
- Guaka-Generator vendors Swiftline for now. This will be resolved when we merge `swiftline/linux!` branch in swiftline.

cc @goyox86

### CODE REVIEW

**Comment by nsomar** [comment]
New command looks like this:

image

Add command:

image

Some Errors:

image

image

File: Makefile [hunk 1]
```diff
@@ -93,6 +93,15 @@ release-publish-local:
 	make release-local
 	make publish-local
 
+clone-swiftline:
+	@echo "Removing old Swiftline"
+	@echo ""
+	rm -rf SwiftLineTemp
+
+	@echo "Clone new Swiftline from oarrabi/linux! branch"
+	git clone -b oarrabi/linux! https://github.com/oarrabi/Swiftline.git SwiftLineTemp
```
Comment (goyox86) [inline Makefile:102]: Are we commiting to maintaing this branch green. I think we shoudl use another name. Like 'vendored'.

File: Sources/GuakaClILib/MiscTypes.swift [hunk 2]
```diff
@@ -20,35 +22,53 @@ public enum GuakaError: Error {
...
   case wrongCommandNameFormat(String)
 
   public var error: String {
     switch self {
     case .wrongDirectoryGiven(let path):
-      return "The path given cannot be used \(path)"
+      return [
+        "Wrong path given:",
+        "  \(path)",
+        "The path cannot be used as its not empty"
```
Comment (goyox86) [inline Sources/GuakaClILib/MiscTypes.swift]: "The path must be an empty directory"

File: Sources/GuakaClILib/MiscTypes.swift [hunk 3]
```diff
@@ -20,35 +22,53 @@ public enum GuakaError: Error {
...
-      return "Cannot generate \(name) file"
+      return "Cannot generate \(name) file".f.red
+
     case .cannotReadFile(let path):
-      return "Cannot read contents of file \(path)"
+      return "Cannot read contents of file \(path)".f.red
+
     case .setupFileAltered:
-      return "Guaka setup.swift file has been altered.\nThe placeholder used to insert commands cannot be found \(GeneratorParts.comamndAddingPlaceholder).\nYou can try to add it yourself by updating `setup.swift` to look like\n\n\(GeneratorParts.setupFileContent())\n\nAdding command wont be possible."
+      return "Guaka setup.swift file has been altered.\nThe placeholder used to insert commands cannot be found \(GeneratorParts.comamndAddingPlaceholder).\nYou can try to add it yourself by updating `setup.swift` to look like\n\n\(GeneratorParts.setupFileContent())\n\nAdding command wont be possible.".f.red
```
Comment (goyox86) [inline Sources/GuakaClILib/MiscTypes.swift]: "Adding command won't be possible"

File: Sources/GuakaClILib/MiscTypes.swift [hunk 4]
```diff
@@ -20,35 +22,53 @@ public enum GuakaError: Error {
...
+
     case .notAGuakaProject:
-      return "This command can only be executed in a Guaka project.\nThe following directory does not contain guaka files"
+      return "This command can only be executed in a Guaka project.\nThe following directory does not contain guaka files".f.red
```
Comment (goyox86) [inline Sources/GuakaClILib/MiscTypes.swift]: "This command can only be executed inside a Guaka project." I think it's enough.

File: Sources/GuakaClILib/MiscTypes.swift [hunk 5]
```diff
@@ -20,35 +22,53 @@ public enum GuakaError: Error {
...
+
     case .missingCommandName:
       return [
-        "`guaka add` requires a command that was not given.",
+        "`guaka add` requires a command that was not given.".f.red,
```
Comment (goyox86) [inline Sources/GuakaClILib/MiscTypes.swift]: "Missing sub-command for `guaka add`"?

File: Sources/GuakaClILib/MiscTypes.swift [hunk 6]
```diff
@@ -20,35 +22,53 @@ public enum GuakaError: Error {
...
+        "",
         "Call `guaka add CommandName` to create a new command.",
         ""
         ].joined(separator: "\n")
 
+    case .commandAlreadyExist(let name, let path):
+      return [ "The command name passed `\(name)` already exist:".f.red,
```
Comment (goyox86) [inline Sources/GuakaClILib/MiscTypes.swift]: "The command  `\(name)` already exists:"

File: Sources/GuakaClILib/MiscTypes.swift [hunk 7]
```diff
@@ -57,8 +77,10 @@ public enum GuakaError: Error {
         "   guaka new my-command",
         "   guaka new my_command",
         "   guaka new myCommand"].joined(separator: "\n")
+
     case .tooManyArgsPassed:
-      return "Too many arguments passed to command."
+      return "Too many arguments passed to command.".f.red
```
Comment (goyox86) [inline Sources/GuakaClILib/MiscTypes.swift:82]: Can't we just ignore the extra args? Also we could say something like "Too many arguments. Trailing arguments will be ignored"

File: Makefile (hunk 1 above)
Comment (nsomar) [reply Makefile:102]: The branch in Swiftline is about making it work on linux https://github.com/oarrabi/Swiftline/pull/30

I think we should keep the name until we merge it

File: Sources/GuakaClILib/MiscTypes.swift (hunk 2 above)
Comment (nsomar) [reply Sources/GuakaClILib/MiscTypes.swift]: Done

File: Sources/GuakaClILib/MiscTypes.swift (hunk 3 above)
Comment (nsomar) [reply Sources/GuakaClILib/MiscTypes.swift]: Done

File: Sources/GuakaClILib/MiscTypes.swift (hunk 5 above)
Comment (nsomar) [reply Sources/GuakaClILib/MiscTypes.swift]: Done

File: Sources/GuakaClILib/MiscTypes.swift (hunk 4 above)
Comment (nsomar) [reply Sources/GuakaClILib/MiscTypes.swift]: Done

File: Sources/GuakaClILib/MiscTypes.swift (hunk 7 above)
Comment (nsomar) [reply Sources/GuakaClILib/MiscTypes.swift:82]: Can we scope this out of this PR? I created #6 for that

**Comment by goyox86** [comment]
👍

**Pull request merged**